google/cloud/spanner/internal/polling_loop.h

```diff
@@ -28,6 +28,65 @@ namespace cloud {
 namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
 namespace internal {
+/**
+ * Extract the result of a long-running operation from the `response` field.
+ */
+template <typename ResultType>
+struct PollingLoopResponseExtractor {
+  using ReturnType = StatusOr<ResultType>;
+
+  static ReturnType Extract(google::longrunning::Operation const& operation,
+                            char const* location) {
```

Comment:
What is the `location` argument, and why is it named "location"? I see it used in some Status error messages, so it looks like it could be any string. But what's the intended meaning of location?

Reply:
Typically the `__func__` that started the loop, that way we report something like "error in UpdateDatabase" instead of "error in PollingLoop for mystery function".